hazelcast/src/main/java/com/hazelcast/util/executor/StripedExecutor.java

```diff
@@ -208,6 +208,7 @@ public void run() {
                         Runnable task = workQueue.take();
                         process(task);
                     } catch (InterruptedException e) {
+                        currentThread().interrupt();
```

Comment:
Same comment as above can be applied here. The thread will exit regardless of the interrupt flag being restored.